replica/database.hh

```diff
@@ -1116,13 +1115,24 @@ public:
 //      - Time Window strategy has one group per window
 class compaction_group {
     table& _t;
+    // SSTable set which contains all non-maintenance sstables
+    lw_shared_ptr<sstables::sstable_set> _main_sstables;
     std::unique_ptr<table::table_state> _table_state;
 public:
     compaction_group(table& t);
 
     // Will stop ongoing compaction on behalf of this group, etc.
     future<> stop();
 
+    // Clear sstable set content, etc.
+    void clear();
+
+    // Add sstable to main set
+    void add_sstable(sstables::shared_sstable sstable);
+
+    const lw_shared_ptr<sstables::sstable_set>& main_sstables() const;
+    void update_main_sstables(lw_shared_ptr<sstables::sstable_set> new_set);
```

Comment:
nit: rename to `set_main_sstables`. `update` may imply the new_set is added into the existing set rather than replacing it.

Reply:
Makes sense. thanks!